olive/cache.py

```diff
@@ -148,6 +145,21 @@ def get_non_local_resource(resource_path: ResourcePath, cache_dir: Union[str, Pa
     return local_resource_path
 
 
+def get_local_path(resource_path: ResourcePath, cache_dir: Union[str, Path] = ".olive-cache"):
+    """
+    Return the local path of the any resource path.
+    If the resource path is a local resource, the path is returned.
+    If the resource path is an AzureML resource, the resource is downloaded to the cache and the path is returned.
+    """
+    if resource_path is None:
+        return None
+
+    if resource_path.is_local_resource() or resource_path.is_string_name():
+        return resource_path.get_path()
+    elif resource_path.is_azureml_resource():
+        return get_non_local_resource(resource_path).get_path()
```

Comment:
This is missing the cache_dir argument. 

We might have to add `.olive_cache` to gitignore since this cache is different from the run cache directory `cache`. Maybe we can find a way to propagate the cache from the run to other parts? Like setting an environment variable like `OLIVE_CACHE_DIR`? And make the default value for cache_dir be `.olive_cache` only if `OLIVE_CACHE_DIR` is not set.